tests/unit/test_charm.py

```diff
@@ -775,298 +790,282 @@ def test_validate_config_options(self, _charm_lib):
-        assert self.harness.charm.get_secret(scope, "somekey") is None
-
-    @pytest.mark.usefixtures("use_caplog")
-    @patch_network_get(private_address="1.1.1.1")
-    @patch("charm.PostgresqlOperatorCharm._on_leader_elected")
-    def test_delete_password(self, _):
-        """NOTE: currently ops.testing seems to allow for non-leader to remove secrets too!"""
-        self.harness.set_leader(True)
-        self.harness.update_relation_data(
-            self.rel_id, self.charm.app.name, {"replication": "somepw"}
-        )
-        self.harness.charm.remove_secret("app", "replication")
-        assert self.harness.charm.get_secret("app", "replication") is None
+        harness.charm.set_secret(scope, "new-secret2", "blablabla")
+        assert harness.charm.get_secret(scope, "new-secret2") == "blablabla"
 
-        self.harness.set_leader(False)
-        self.harness.update_relation_data(
-            self.rel_id, self.charm.unit.name, {"somekey": "somevalue"}
-        )
-        self.harness.charm.remove_secret("unit", "somekey")
-        assert self.harness.charm.get_secret("unit", "somekey") is None
 
-        self.harness.set_leader(True)
-        with self._caplog.at_level(logging.ERROR):
-            self.harness.charm.remove_secret("app", "replication")
-            assert (
-                "Non-existing field 'replication' was attempted to be removed" in self._caplog.text
-            )
+@pytest.mark.parametrize("scope,is_leader", [("app", True), ("unit", True), ("unit", False)])
+@patch_network_get(private_address="1.1.1.1")
+def test_invalid_secret(harness, scope, is_leader):
+    with (
+        patch("charm.PostgresqlOperatorCharm._on_leader_elected"),
+    ):
+        # App has to be leader, unit can be either
+        harness.set_leader(is_leader)
 
-            self.harness.charm.remove_secret("unit", "somekey")
-            assert "Non-existing field 'somekey' was attempted to be removed" in self._caplog.text
+        with tc.assertRaises(RelationDataTypeError):
+            harness.charm.set_secret(scope, "somekey", 1)
 
-            self.harness.charm.remove_secret("app", "non-existing-secret")
-            assert (
-                "Non-existing field 'non-existing-secret' was attempted to be removed"
-                in self._caplog.text
-            )
+        harness.charm.set_secret(scope, "somekey", "")
+        assert harness.charm.get_secret(scope, "somekey") is None
 
-            self.harness.charm.remove_secret("unit", "non-existing-secret")
-            assert (
-                "Non-existing field 'non-existing-secret' was attempted to be removed"
-                in self._caplog.text
-            )
 
-    @patch("charm.JujuVersion.has_secrets", new_callable=PropertyMock, return_value=True)
-    @patch_network_get(private_address="1.1.1.1")
-    @patch("charm.PostgresqlOperatorCharm._on_leader_elected")
-    @pytest.mark.usefixtures("use_caplog")
-    def test_delete_existing_password_secrets(self, _, __):
```

Comment:
similar to what was done in https://github.com/canonical/postgresql-operator/pull/425, this test got removed and `test_delete_password` runs with and without secrets